controllers/kyma_controller.go

```diff
@@ -231,48 +232,20 @@ func (r *KymaReconciler) HandleInitialState(ctx context.Context, kyma *v1alpha1.
 func (r *KymaReconciler) HandleProcessingState(ctx context.Context, kyma *v1alpha1.Kyma) error {
 	logger := log.FromContext(ctx)
 
-	statusUpdateRequiredFromCatalog, err := r.syncModuleCatalog(ctx, kyma)
-	if err != nil {
-		return r.UpdateStatusWithEventFromErr(ctx, kyma, v1alpha1.StateError,
-			fmt.Errorf("could not synchronize remote module catalog: %w", err))
-	}
-
-	var modules common.Modules
-	// these are the actual modules
-	modules, err = r.GenerateModulesFromTemplate(ctx, kyma)
-	if err != nil {
-		return r.UpdateStatusWithEventFromErr(ctx, kyma, v1alpha1.StateError,
-			fmt.Errorf("error while fetching modules during processing: %w", err))
-	}
-
-	runner := sync.New(r)
-
-	statusUpdateRequiredFromModuleSync, err := runner.Sync(ctx, kyma, modules)
+	statusUpdateRequiredFromModulesSync, err := r.syncModules(ctx, kyma)
```

Comment:
any particular reason to move this out?

Reply:
I moved this part to a separate method because of the cyclomatic complexity linting error. see https://status.build.kyma-project.io/view/gs/kyma-prow-logs/pr-logs/pull/kyma-project_lifecycle-manager/313/pull-lifecycle-mgr-lint/1613491447321858048